test/test-thread-affinity.c

```diff
@@ -0,0 +1,63 @@
+/* Copyright libuv project contributors. All rights reserved.
+ */
+
+#include "uv.h"
+#include "task.h"
+
+#include <string.h>
+
+static void check_affinity(void* arg) {
+  int r;
+  char* cpumask = arg;
+  int cpumasksize = uv_cpumask_size();
+
+  uv_thread_t tid = uv_thread_self();
+  uv_thread_setaffinity(&tid, cpumask, NULL, cpumasksize);
+  r = uv_thread_setaffinity(&tid, cpumask + cpumasksize, cpumask, cpumasksize);
+  if (r != 0)
+    cpumask[0] = cpumask[1] = -1;
+}
+
+TEST_IMPL(thread_affinity) {
```

Comment:
On platforms that don't support this we should test that `uv_cpumask_size` returns `UV_ENOTSUP` instead:

```c
#if defined(__APPLE__) && defined(__MACH__) || defined(_AIX)
  ASSERT(UV_ENOTSUP == uv_cpumask_size());
  return 0;
#else
...
```